### PR TITLE
annotation support

### PR DESCRIPTION
**Pull request by xcokazaki**
- Add `Annotation` .
- Add annotation interface to `Layer` .
- Add few tests.

### CODE REVIEW

**Comment by xcokazaki** [comment]
Sorry I made wrong PR on this repo. Please ignore this.

**Comment by Vindaar** [comment]
@xcokazaki Do you plan to reopen this / make another PR with annotation support? In principle that's a useful feature, I think. I haven't looked at annotations in plotly yet, but it's something that I'll need soon anyway.

**Comment by xcokazaki** [comment]
@Vindaar Yes, I can reopen this. I just wanted to achieve [annotated_heatmap](https://plot.ly/python/annotated_heatmap/) feature. As I'm not so familiar with nim(actually I'm a newbie), what should I do for this PR? Should I add any examples for this feature?

**Comment by brentp** [comment]
I agree, this looks useful. An example would be nice, but the tests are great!

**Comment by xcokazaki** [comment]
OK, then I'll reopen this PR. I think I need tiny modifications on this. When it's ready, please review the codes.

**Comment by xcokazaki** [comment]
Now it's ready to review.

File: tests/plotly/test_api.nim [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                                        , "text": "text"
+                                        , "showarrow": false
+                                        }
+                                      ]
+                     }
+    let r = %layout
+    check r == expected
+  test "Layout without Annotations":
+    let
+        a = Annotation(x:1, xshift:10, y:2, yshift:20, text:"text")
```
Comment (brentp) [inline tests/plotly/test_api.nim]: can remove `a` here, yes?

Comment (xcokazaki) [reply tests/plotly/test_api.nim]: Oh, right! I'll delete this.

**Pull request merged**